### ISSUE TITLE
report multiple tests with the same results at the same time

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4487]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4487|
|Type|enhancement|
|Created|2010-06-04 17:09:23Z|
|Branch|https://github.com/twisted/twisted/tree/trial-shorter-summary-4487|

If there are several tests which are skipped (or todo'd, or fail) for the same reason, trial reports them each separately, duplicating the reason each time.

It would be nicer if it grouped together these reports by reason, so that any particular reason would only be reported once, and all of the tests affected by it would be collected together nearby.

Attachments:

* [trial.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/282/2824f731f7afff8e49d5682a5af9773b920c0618/37c388d26c369b2908bf3e70d7169ecc7215e792.txt) (40401 bytes) - added by exarkun on 2010-06-29 22:29:23Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4487 4487
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_trial_shorter_summary_4487 branches/trial-shorter-summary-4487
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1275671363000000 1275671363000000
changetime__1278774319000000 1278774319000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29464]) Branching to 'trial-shorter-summary-4487'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Implemented something.  Attached is an example of the new output when running Twisted's test suite.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/trial-shorter-summary-4487)

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|Screwtape set owner to @exarkun|
|-|-|

I ran the tests with every reporter that works on my system, and it prints nicely summarized output every time (and no output at all for "summary", as it should). I couldn't test "subunit", but I don't know any reason why it would break when everything else works.

+1 for merging from me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29761]) Merge trial-shorter-summary-4487 

Author: exarkun
Reviewer: Screwtape
Fixes: [#4487](https://github.com/twisted/twisted/issues/4487)

Most trial reporters now reports multiple tests with the same results in a combined
display in the summary section, shorting the overall output.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
